android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/translate/TranslationErrorDisplay.kt

```diff
@@ -0,0 +1,54 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.concept.engine.translate
+
+/**
+ * The types of translation errors we display to the user.
+ *
+ * @param errorName The error name of the translation error to display.
+ */
+enum class TranslationErrorDisplay(val errorName: String) {
```

Comment:
Maybe we could transform `TranslationError` into a [sealed class](https://kotlinlang.org/docs/sealed-classes.html), so we could have a generic `TranslationError` with multiple children or groups : 


``` kotlin
sealed class TranslationError(val errorName: String) {
 data class NoDisplay(override val caller: errorName) : TranslationError()
 data class UnableToTranslate(override val caller: errorName) : TranslationError()
 data class UnableToLoadLanguage(override val caller: errorName) : TranslationError()
 data class LanguageNotSupported(override val caller: errorName) : TranslationError()
}

```

Reply:
What do you think?

Reply:
Also what is the purpose of `errorName`? Is it going to be used for showing a message to the user?

Reply:
Thanks for taking a look!

Sounds like a great idea, I'll give it a try and update!

>Also what is the purpose of errorName?  

Envisioning it as what telemetry will report as the error reason. Something like [how this is used](https://searchfox.org/mozilla-mobile/rev/b4fff7f0b3d73d23a8baf7711dc8e0163518d2a1/firefox-android/fenix/app/metrics.yaml#567-570) or a pre-specified value. (I did the mapping incorrectly in the past like [this](https://searchfox.org/mozilla-mobile/rev/b4fff7f0b3d73d23a8baf7711dc8e0163518d2a1/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/share/SaveToPDFMiddleware.kt#107).)

Reply:
(Could probably just have had `errorName` on `TranslationError` and not needed on `TranslationErrorDisplay`, now that you mention it, though. I'll fix in the refactor at any rate. 😅)

Reply:
Refactored to a sealed class. 🦭 Please let me know if you think this works! I inherited from `Throwable`. 

I am thinking in [bug 1872703](https://bugzilla.mozilla.org/show_bug.cgi?id=1872703) I'll go through to all the translations functions, for example, [`isTranslationsEngineSupported`](https://searchfox.org/mozilla-mobile/rev/bba84847ffbb25a9865b21912792b4186fe13c09/firefox-android/android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt#698) and change the `onError` something like `onError(GeckoTranslationUtils.fromGeckoErrorToTranslationError(throwable))`. 

I might also add another named error for unexpected nulls instead of using [`translationsUnexpectedNull`](https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt#676) to use in that bug.

Reply:
The above sounds good to me.